crates/error/src/cairo_compile.rs

```diff
@@ -0,0 +1,22 @@
+//! Error types and utilities to do with the compilation from Cairo to
+//! [`cairo_lang_lowering::FlatLowered`].
```

Comment:
Yeah I'm not happy with this organization. Please change it to be `hieratika_errors::compile::llvm` and `hieratika_errors::compile::cairo`.